portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java

```diff
@@ -423,6 +424,62 @@ public void clearUserRoles(long userId) throws PortalException {
 		reindex(userId);
 	}
 
+	@Override
+	public Role copyRole(
+			Role sourceRole, String duplicateRoleName,
+			ServiceContext serviceContext)
+		throws PortalException {
+
+		Role duplicateRole = addRole(
+			null, sourceRole.getUserId(), sourceRole.getClassName(), 0,
+			duplicateRoleName, sourceRole.getTitleMap(),
```

Comment:
not sure that the titleMap should be the one of the original role, doesn't it depend on the name? Could you please check how is generated in the add method?